File: 2462-total-cost-to-hire-k-workers/2462-total-cost-to-hire-k-workers.js
public class Solution {
    public long TotalCost(int[] c, int k, int can)
    {
        long ans = 0;
        if (c.Length == 1)
            return c[0];

        PriorityQueue<int,int> l = new();
        PriorityQueue<int,int> r = new();
        int left = 0, right = c.Length - 1;

        for (int i = 1; i <= can; i++)
        {
            if (i - 1 >= c.Length - i)
                break;

            l.Enqueue(c[i - 1], c[i - 1]);
            left = i;
            r.Enqueue(c[c.Length - i], c[c.Length - i]);
            right = c.Length - i - 1;
        }

        bool f = true;
        while (k > 0)
        {
            if (left > right)
                f = false;

            int a = l.Count>0 ? l.Peek() : int.MaxValue;
            int b = r.Count>0 ? r.Peek() : int.MaxValue;

            if (a <= b)
            {
                ans += a;
                l.Dequeue();
                if (f)
                    l.Enqueue(c[left],c[left]);
                if (f)
                    left++;
            }
            else
            {
                ans += b;
                r.Dequeue();
                if (f)
                    r.Enqueue(c[right],c[right]);
                if (f)
                    right--;
            }

            k--;
        }

        return ans;
    }
}

File: 2462-total-cost-to-hire-k-workers/2462-total-cost-to-hire-k-workers.kt
class Solution {
    fun totalCost(costs: IntArray, k: Int, candidates: Int): Long {
        val pqL = PriorityQueue<Int>()
        val pqR = PriorityQueue<Int>()
        var lo = 0
        var hi = costs.lastIndex
        var sum = 0L
        var count = 0
        if (2 * candidates >= costs.size) while (lo <= hi) pqL.add(costs[lo++])
        while (pqL.size < candidates && lo <= hi) pqL.add(costs[lo++])
        while (pqR.size < candidates && lo < hi) pqR.add(costs[hi--])
        while (lo <= hi && count++ < k) {
            if (pqR.peek() < pqL.peek()) {
                sum += pqR.poll()
                pqR.add(costs[hi--])
            } else {
                sum += pqL.poll()
                pqL.add(costs[lo++])
            }
        }
        while (pqR.isNotEmpty()) pqL.add(pqR.poll())
        while (count++ < k && pqL.isNotEmpty()) sum += pqL.poll()
        return sum
    }
}